tasks/main.yml

```diff
@@ -10,11 +10,32 @@
     enabled: true
     state: 'started'
 
+- name: Check if firewalld is installed
+  command:
+  args:
+    cmd: 'rpm -q firewalld'
+    warn: false
+  register: check_for_firewalld
+  failed_when: false
+  changed_when: false
+  check_mode: false
+
+- name: Check if firewalld is running
+  command:
+  args:
+    cmd: 'systemctl is-active firewalld'
+    warn: false
+  register: check_for_running_firewalld
+  failed_when: false
+  changed_when: false
+  check_mode: false
+
 - name: Enable port 3260/tcp on firewalld
   firewalld:
     port: '3260/tcp'
     permanent: true
     state: 'enabled'
+  when: '"firewalld is not installed" not in check_for_firewalld.stdout and "inactive" not in check_for_running_firewalld.stdout'
```

Comment:
I would avoid testing one reason where this will not work and rather focus on when only this makes sense. For example it makes sense to run this module only when "firewalld.service is active". If the firewalld.service is active, then it is most probably installed (chance that your system has it running and it is not installed is quite small).
Could you please simplify this to check only for "running firewalld.service"?

Reply:
see new commit b067df7